Python/ceval.c

```diff
@@ -2095,8 +2095,15 @@ _PyEval_ExceptionGroupMatch(PyObject* exc_value, PyObject *match_type,
         if (pair == NULL) {
             return -1;
         }
-        assert(PyTuple_CheckExact(pair));
-        assert(PyTuple_GET_SIZE(pair) == 2);
+
+        if (!PyTuple_CheckExact(pair) || PyTuple_Size(pair) != 2) {
+            PyErr_Format(PyExc_TypeError,
+                        "%s.split must return a 2-tuple",
+                        Py_TYPE(exc_value)->tp_name);
```

Comment:
Does this work?

```suggestion
            PyErr_Format(PyExc_TypeError,
                         "invalid type returned from %s.split: expected a 2-tuple, got %s",
                        Py_TYPE(exc_value)->tp_name, Py_TYPE(pair)->tp_name);
```

Reply:
If we want the error messages to be more descriptive, we should probably split up the checks. It would be a bit confusing to see the message
```
invalid type returned from EG.split: expected a 2-tuple, got tuple
```
if you had returned a tuple of length 1

Reply:
I like the idea of having more descriptive error messages, I was thinking something like this
```c
if (!PyTuple_CheckExact(pair))
{
    PyErr_Format(PyExc_TypeError,
                "%.200s.split must return a tuple, not %.200s",
                Py_TYPE(exc_value)->tp_name, Py_TYPE(pair)->tp_name);
    Py_DECREF(pair);
    return -1;
}

if (PyTuple_GET_SIZE(pair) < 2) {
    PyErr_Format(PyExc_TypeError,
                "%.200s.split must return a 2-tuple, got tuple of size %zd",
                Py_TYPE(exc_value)->tp_name, PyTuple_GET_SIZE(pair));
    Py_DECREF(pair);
    return -1;
}
```

Reply:
Ok. Should it be `%.200s.split()`?

Reply:
I personally don't think so since the func takes in an arg and that makes it look like it doesnt